### PR TITLE
additional debugger features

### PR DESCRIPTION
**Pull request by Zychronix**
auto advance text, fast text speed, "auto" charm potion, fix combat debugger not displaying full HP stat

### CODE REVIEW

File: src/main/java/legend/game/SMap.java [hunk 1]
```diff
@@ -3878,12 +3879,17 @@ public static long handleEncounters() {
       return 0;
     }
 
-    encounterAccumulator_800c6ae8.add(encounterData_800f64c4.get(submapCut_80052c30.get()).rate_02.get() * encounterMultiplier_800c6abc.get());
+    if (!Config.autoCharmPotion()) {
```
Comment (LordMonoxide) [inline src/main/java/legend/game/SMap.java:3882]: CS: no space after if

File: src/main/java/legend/game/SMap.java [hunk 2]
```diff
@@ -3878,12 +3879,17 @@ public static long handleEncounters() {
...
+      encounterAccumulator_800c6ae8.add(encounterData_800f64c4.get(submapCut_80052c30.get()).rate_02.get() * encounterMultiplier_800c6abc.get());
+    }
 
     if(encounterAccumulator_800c6ae8.get() > 0x1400) {
       // Start combat
       encounterId_800bb0f8.set(sceneEncounterIds_800f74c4.get(encounterData_800f64c4.get(submapCut_80052c30.get()).scene_00.get()).get(randomEncounterIndex()).get());
       combatStage_800bb0f4.set(encounterData_800f64c4.get(submapCut_80052c30.get()).stage_03.get());
+      if (Config.combatStage()) {
```
Comment (LordMonoxide) [inline src/main/java/legend/game/SMap.java:3890]: CS: no space after if

File: src/main/java/legend/core/Config.java [hunk 3]
```diff
@@ -173,4 +225,39 @@ public static void load() throws IOException {
   public static void save() throws IOException {
     properties.store(Files.newOutputStream(path, StandardOpenOption.WRITE, StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING), "");
   }
+
+
+  static class SortedStoreProperties extends Properties {
+
+    @Override
+    public void store(OutputStream out, String comments) throws IOException {
```
Comment (LordMonoxide) [inline src/main/java/legend/core/Config.java:233]: CS: final all vars possible

**Review by LordMonoxide — CHANGES_REQUESTED** [review]

File: src/main/java/legend/game/SMap.java (hunk 1 above)
Comment (Zychronix) [reply src/main/java/legend/game/SMap.java:3882]: fixed

File: src/main/java/legend/game/SMap.java (hunk 2 above)
Comment (Zychronix) [reply src/main/java/legend/game/SMap.java]: fixed

File: src/main/java/legend/core/Config.java (hunk 3 above)
Comment (Zychronix) [reply src/main/java/legend/core/Config.java]: still works with final, fixed

**Review by LordMonoxide — APPROVED** [review]

**Pull request merged**